component/src/main/ballerina/ftp/client_action.bal

```diff
@@ -0,0 +1,114 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+
+documentation {
+    Provides the FTP client actions for interacting with an FTP server.
+
+    F{{config}} The configurations of the client endpoint associated with this ClientActions instance.
+}
+public type ClientActions object {
+    public {
+        ClientEndpointConfiguration config;
+    }
+
+    documentation {
+        The `get()` function can be used to retrieve file content from a remote resource.
+
+        P{{path}} The resource path
+        R{{}} A ByteChannel which represent the data source to the resource or
+        an `error` if failed to establish communication with the FTP server or read the resource
+    }
+    public native function get(string path) returns io:ByteChannel|error;
+
+    documentation {
+        The `delete()` function can be used to delete a file from an FTP server.
+
+        P{{path}} The resource path
+        R{{}} Returns an `error` if failed to establish communication with the FTP server
+    }
+    public native function delete(string path) returns error?;
+
+    documentation {
+        The `put()` function can be used to add a file to an FTP server.
+
+        P{{path}} The resource path
+        P{{channel}} A ByteChannel that represent the local data source
+        R{{}} Returns an `error` if failed to establish communication with the FTP server
+    }
+    public native function put(string path, io:ByteChannel channel) returns error?;
+
+    documentation {
+        The `append()` function can be used to append content to an existing file in an FTP server.
+        A new file will create if file does not exist.
+
+        P{{path}} The resource path
+        P{{channel}} A ByteChannel that represent the local data source
+        R{{}} Returns an `error` if failed to establish communication with the FTP server
+    }
+    public native function append(string path, io:ByteChannel channel) returns error?;
+
+    documentation {
+        The `mkdir()` function can be used to create new direcotry in an FTP server.
+
+        P{{path}} The directory path
+        R{{}} Returns an `error` if failed to establish communication with the FTP server
+    }
+    public native function mkdir(string path) returns error?;
+
+    documentation {
+        The `rmdir()` function can be used to delete an empty direcotry in an FTP server.
+
+        P{{path}} The directory path
+        R{{}} Returns an `error` if failed to establish communication with the FTP server
+    }
+    public native function rmdir(string path) returns error?;
+
+    documentation {
+        The `rename()` function can be used to rename a file or move to new location with in the same FTP server.
```

Comment:
Made minor fixes. Use: 'The `rename()` function can be used to rename a file or move to a new location within the same FTP server.'